README.md
# Basic PHP 7 eCommerce Website

A basic PHP 7 website which includes MySQL database and Shopping Cart functionality.

###### Table of Contents

<!-- TOC START min:2 max:5 link:true update:true -->
- [Getting Started](#getting-started)
  - [Prerequisites](#prerequisites)
  - [Installing](#installing)
- [Deployment](#deployment)
- [Contributing](#contributing)
- [Code of Conduct](#code-of-conduct)
- [Versioning](#versioning)
- [Change-Log](#change-log)
- [Authors](#authors)
- [License](#license)
- [Acknowledgements](#acknowledgements)

<!-- TOC END -->

## Getting Started

These instructions will get your copy of the project up and running on your
local machine for development and testing purposes. [See deployment][deploy]
for notes on how to deploy the project on a live system.

### Prerequisites

You will need the following running on your local development machine.

- Apache or Nginx
- PHP 7.*
- MySQL or MariaDB

### Installing

A step-by-step series of examples that tell you how to get a development
environment running.

- [ ] Add Installing steps.

## Deployment

- [ ] Add deployment notes.

## Contributing

Please read the [CONTRIBUTING.md][CONTRIBUTING] file for details on how you
can get involved in the project as well as the process for submitting bugs
and pull requests.

## Code of Conduct

Please read the [CODE_OF_CONDUCT.md][COC] file for the guidelines that govern
the community.

## Versioning

We use [Semantic Versioning][semver] for software versions of this project.
For a list of all the versions available, see the [tags][tags] and
[releases][releases] on this repository.

## Change-Log

View the [CHANGELOG.md][changelog] file for a detailed list of changes,
along with specific tasks completed for each version released to date.

## Authors

- Justin Hartman - [@justinhartman][author-1]

Also see the list of [contributors][contribs] who have participated in this
project.

## License

This project is licensed under the `GNU Affero General Public License` License. 
See the [LICENSE][license] file for full details.

## Acknowledgements

Special thanks go out to the following people and projects who have helped in
some way to make this project a reality.

- [@justinhartman/.github][.github] - for the Github project templates.

[deploy]: #deployment
[CONTRIBUTING]: CONTRIBUTING.md
[COC]: CODE_OF_CONDUCT.md
[license]: LICENSE
[changelog]: CHANGELOG.md
[semver]: http://semver.org
[tags]: https://github.com/justinhartman/basic-php7-ecom-site/tags
[releases]: https://github.com/justinhartman/basic-php7-ecom-site/releases
[contribs]: https://github.com/justinhartman/basic-php7-ecom-site/contributors
[author-1]: https://github.com/justinhartman
[.github]: https://github.com/justinhartman/.github


CHANGELOG.md
# Project Change-Log

Below is a detailed change-log, along with specific tasks completed, for each 
version released to date.

## Version 0.1.0 (17/10/2018)

- [#new](#new)
  - New template files for:
    - `navigation.php`
    - `header.php`
    - `footer.php`
  - New core files for:
    - `index.php`
    - `cart.php`
    - `add_to_cart.php`
    - `delete_from_cart.php`
- [#enhancement](#enhancement)
  - Added GitHub template files.
  - Updated the `README.md` file with project specific information.
  - New `.editorconfig` file.
- [#bugfix](#bugfix)
  - Updated GitHub template files with project details.

## Version 0.0.1 (17/10/2018)

- [#new](#new)
  - First Commit to the Repo.


cart.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
session_start();
require_once('inc/connect.php');
include('templates/header.php');
include('templates/nav.php');
?>

<div class="container">
    <?php
        $items = $_SESSION['cart'];
        $cartitems = explode(",", $items);
    ?>
    <div class="row">
        <table class="table">
            <tr>
                <th>S.NO</th>
                <th>Item Name</th>
                <th>Price</th>
            </tr>
        <?php
            $total = '';
            $i=1;
            foreach ($cartitems as $key=>$id) {
                $sql = "SELECT * FROM products WHERE id = $id";
                $res=mysqli_query($connection, $sql);
                $r = mysqli_fetch_assoc($res); ?>
            <tr>
                <td><?php echo $i; ?></td>
                <td><a href="delete_from_cart.php?remove=<?php echo $key; ?>">Remove</a> <?php echo $r['title']; ?></td>
                <td>$<?php echo $r['price']; ?></td>
            </tr>
            <?php
                $total = $total + $r['price'];
                $i++;
            } ?>
            <tr>
                <td><strong>Total Price</strong></td>
                <td><strong>R<?php echo $total; ?></strong></td>
                <td><a href="#" class="btn btn-info">Checkout</a></td>
            </tr>
        </table>
    </div>
</div>

<?php include('templates/footer.php'); ?>


index.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
session_start();
require_once('inc/connect.php');
include('templates/header.php');
include('templates/nav.php');

$sql = "SELECT * FROM products";
$res = mysqli_query($connection, $sql);
?>

<div class="container">
    <?php
        if (isset($_GET['status']) & !empty($_GET['status'])) {
            if ($_GET['status'] == 'success') {
                echo "<div class=\"alert alert-success\" role=\"alert\">Item Successfully Added to Cart.</div>";
            } elseif ($_GET['status'] == 'incart') {
                echo "<div class=\"alert alert-info\" role=\"alert\">Item Already Exists in Cart!</div>";
            } elseif ($_GET['status'] == 'failed') {
                echo "<div class=\"alert alert-danger\" role=\"alert\">Failed to add item, please try again!</div>";
            }
        }
    ?>
    <div class="row">
        <?php while ($r = mysqli_fetch_assoc($res)) { ?>
            <div class="col-sm-6 col-md-3">
                <div class="thumbnail">
                    <img src="<?php echo $r['image']; ?>" alt="<?php echo $r['title'] ?>">
                    <div class="caption">
                        <h3><?php echo $r['title'] ?></h3>
                        <p><?php echo $r['description'] ?></p>
                        <p><a href="add_to_cart.php?id=<?php echo $r['id']; ?>" class="btn btn-primary" role="button">Add to Cart</a></p>
                    </div>
                </div>
            </div>
        <?php } ?>
    </div>

</div>

<?php include('templates/footer.php'); ?>


add_to_cart.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
session_start();

if (isset($_GET['id']) & !empty($_GET['id'])) {
    if (isset($_SESSION['cart']) & !empty($_SESSION['cart'])) {
        $items = $_SESSION['cart'];
        $cartitems = explode(",", $items);
        if (in_array($_GET['id'], $cartitems)) {
            header('location: index.php?status=incart');
        } else {
            $items .= "," . $_GET['id'];
            $_SESSION['cart'] = $items;
            header('location: index.php?status=success');
        }
    } else {
        $items = $_GET['id'];
        $_SESSION['cart'] = $items;
        header('location: index.php?status=success');
    }
} else {
    header('location: index.php?status=failed');
}


delete_from_cart.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
session_start();
$items = $_SESSION['cart'];
$cartitems = explode(",", $items);
if (isset($_GET['remove']) & !empty($_GET['remove'])) {
    $delitem = $_GET['remove'];
    unset($cartitems[$delitem]);
    $itemids = implode(",", $cartitems);
    $_SESSION['cart'] = $itemids;
}
header('location:cart.php');


templates/footer.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
?>
    </body>
</html>


templates/header.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
?>
<html>
    <head>
        <title>Basic PHP eCommerce Website</title>
        <!-- Latest compiled and minified CSS -->
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" >
        <!-- Optional theme -->
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" >
        <link rel="stylesheet" href="../styles.css" >
        <!-- JQuery -->
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
        <!-- Latest compiled and minified JavaScript -->
        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    </head>

    <body>
        <div class="container">
            <div class="row">
                <div class="col-md-5">
                    <a class="img-responsive logo" href="index.php">
                        <img src="https://via.placeholder.com/276x100" />
                    </a>
                </div>
            </div>
        </div>


templates/navigation.php
<?php
/**
 * Basic PHP 7 eCommerce Website (https://22digital.agency)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 *
 * @copyright Copyright (c) 22 Digital (https://22digital.agency)
 * @copyright Copyright (c) Justin Hartman (https://justinhartman.blog)
 * @author    Justin Hartman <justin@hartman.me> (https://justinhartman.blog)
 * @link      https://github.com/justinhartman/basic-php7-ecom-site GitHub Project
 * @since     0.1.0
 * @license   https://opensource.org/licenses/AGPL-3.0 AGPL-3.0
 */
?>
<nav class="navbar navbar-default">
    <div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
        </div>

        <!-- Collect the nav links, forms, and other content for toggling -->
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav">
                <li><a href="index.php">Shop Home <span class="sr-only">(current)</span></a></li>
                <li><a href="cart.php">Cart</a></li>
            </ul>

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="cart.php" >
                        <?php
                            $items = $_SESSION['cart'];
                            $cartitems = explode(",", $items);
                            echo count($cartitems);
                        ?> Items in Cart
                    </a>
                </li>
            </ul>
        </div><!-- /.navbar-collapse -->
    </div><!-- /.container-fluid -->
</nav>
